app/controllers/application_controller.rb

```diff
@@ -3,6 +3,10 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  before_filter do |c|
+    User.current_user_id = User.find(c.session[:user]).id unless c.session[:user].nil?
+  end
```

Comment:
Such way of sharing request state with model layer can lead to bugs as we use threaded application server. Simply passing user instance is the way to go.